bin/git-theta

```diff
@@ -39,49 +44,64 @@ def parse_args():
     )
     track_parser.set_defaults(func=track)
 
-    args = parser.parse_args()
+    add_parser = subparsers.add_parser(
+        "add", help="add command used to stage files."
+    )
+    add_parser.add_argument("--update-type", choices=["dense", "sparse"], help="Type of update being applied")
+    add_parser.set_defaults(func=add)
+
+    args = parser.parse_known_args()
     return args
 
 
-def paths_and_parameters(model_dict, root):
-    """Convert the keys of the model dict to parameters based starting at root."""
-    return {os.path.join(root, *k): v for k, v in utils.flatten(model_dict).items()}
+def post_commit(args):
+    repo = git_utils.get_git_repo()
+    theta_commits = theta.ThetaCommits(repo)
+
+    gitattributes_file = git_utils.get_gitattributes_file(repo)
+    patterns = git_utils.get_gitattributes_tracked_patterns(gitattributes_file)
+
+    oids = set()
+    commit = repo.commit('HEAD')
+    for obj in commit.tree.traverse():
+        if any([fnmatch.fnmatchcase(obj.path, pattern) for pattern in patterns]):
+            curr_metadata = metadata.Metadata.from_file(obj.data_stream).flatten()
+
+            prev_obj = git_utils.get_file_version(repo, obj.path, 'HEAD~1')
```

Comment:
Does this break with `git mv`? In that case the path at `"HEAD~1"` might be different so `prev_obj` would result in being `None`?

There is a diff index object you get with `diff_index = commit1.diff(commit2)` and the diff objects in it have things like `a_path` and `b_path` which might help us fix this edge case (if it is broken)

Reply:
This is a great point. A couple things definitely break with `git mv`. 

1. post-commit hook uses the metadata file path to figure out what LFS oids were changed in the previous commit (NOTE: this requires looking at the parent or parents of the HEAD commit and inspecting the metadata files)
2. the clean filter uses the path to the metadata file to look at the state of the checkpoint at HEAD
3. the `Update` class uses the path to the metadata file to get the state at a previous commit in `Update.get_last_version`.

Basically, anywhere that we look for a version of the metadata file, we implicitly rely on the fact that the path to it has not changed. We should write up a unified solution for all three places we do this that uses the diff index object you'd mentioned. Since this is a bit tricky, let's push this to another issue.